business/istio_config.go

```diff
@@ -149,59 +157,50 @@ func (in *IstioConfigService) GetIstioConfigDetails(namespace string, objectType
 	istioConfigDetail.ObjectType = objectType
 	var gw, vs, dr, se, qs, qb kubernetes.IstioObject
 	var r *kubernetes.IstioRuleDetails
-	permission := models.ResourcePermissions{}
 	var wg sync.WaitGroup
 	wg.Add(1)
 
 	go func() {
 		defer wg.Done()
-		if api, ok := resourceTypesToAPI[objectType]; ok {
-			ssars, permErr := in.k8s.GetSelfSubjectAccessReview(namespace, api, objectType, []string{"create", "update", "delete"})
-			if permErr == nil {
-				for _, ssar := range ssars {
-					fillPermission(&permission, ssar)
-				}
-			} else {
-				log.Errorf("Error getting permissions [namespace: %s, api: %s, objectType: %s]: %v", namespace, api, objectType, permErr)
-			}
+		canUpdate, canDelete := getUpdateDeletePermissions(in.k8s, namespace, objectType)
+		istioConfigDetail.Permissions = models.ResourcePermissions{
```

Comment:
I wonder if istioConfigDetail will pass the race tests.
As it seems the istioConfigDetail is updated from two goroutines.
But it touches different fields so I guess it's fine.

Reply:
I think it's ok, as you say it's two fields with two different memory addresses